### PR TITLE
Add macOS build and test GitHub workflow script

### PR DESCRIPTION
**Pull request by 030emily**
*Issue #, if available:*
N/A

*Description of changes:*
- Added new GitHub Actions workflow for macOS builds and configured build and test steps for macOS environment
- Includes automated unit testing on macOS runners

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/osx_build.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+jobs:
+  OSX:
+    name: Build on ${{matrix.runner_image}}
+    runs-on: ${{matrix.runner_image}}
+    strategy:
+      matrix:
+        include:
+          - runner_image: macos-13
+            s3_path: X86_64/OSX/MacOS
+            display_name: "MacOS Intel"
```
Comment (13ajay) [inline .github/workflows/osx_build.yml:19]: Minor, but consistency in the format adhered to by the display name might be good to consider. I think the ISA might be better to use as the second word in the display name, so "MacOS x86-64" and "MacOS Aarch64" (I'm using "Aarch64" here since that's the term we use elsewhere generally).

File: .github/workflows/osx_build.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+jobs:
+  OSX:
+    name: Build on ${{matrix.runner_image}}
+    runs-on: ${{matrix.runner_image}}
+    strategy:
+      matrix:
+        include:
+          - runner_image: macos-13
+            s3_path: X86_64/OSX/MacOS
```
Comment (13ajay) [inline .github/workflows/osx_build.yml:18]: "OSX" and "MacOS" are effectively synonymous. Do we need to have both in the S3 path?

File: .github/workflows/osx_build.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            display_name: "MacOS Intel"
+          - runner_image: macos-14
```
Comment (13ajay) [inline .github/workflows/osx_build.yml:20]: What about backwards compatibility? Ideally, we would build on the oldest version that isn't EOL for each architecture, but do we have that level of control here?

File: .github/workflows/osx_build.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+
+      - name: Build Project
+        run: |
+          echo "Building on ${{ matrix.display_name }}"
+          make release
+          make test-certs
+          make test
```
Comment (13ajay) [inline .github/workflows/osx_build.yml:39]: The `test` target already depends on the `test-certs` target, so there's no need to run `test-certs` first explicitly. 

Also, have you been able to test negative scenarios here? If a unit test fails, does that cause the last command to return a non-zero exit code, and that in turn will make it so that the rest of the steps don't run?

**Review by 13ajay — APPROVED** [review]
